typescript/orval.config.ts

```diff
@@ -18,12 +19,13 @@ export default defineConfig({
         mutator: {
           path: "./cdpApiClient.ts",
           name: "cdpApiClient",
+          extension: ".js",
```

Comment:
See: https://github.com/coinbase/cdp-sdk/pull/49/files#r2053000753

I have also included a patch in this PR to patch orval and ensure generated code with `exports` or `imports` have the file extension.